assets/js/src/post_editor_block/edit.jsx

```diff
@@ -1,18 +1,26 @@
 /* eslint-disable react/react-in-jsx-scope */
-const wp = window.wp;
-const { useState } = wp.element;
+import PropTypes from 'prop-types';
 
 const allForms = window.mailpoet_forms;
 
-function Edit() {
+function Edit({ attributes, setAttributes }) {
   function displayFormsSelect() {
     if (!Array.isArray(allForms)) return null;
     if (allForms.length === 0) return null;
     return (
-      <select>
+      <select
+        onChange={(event) => {
+          setAttributes({
+            selectedForm: parseInt(event.target.value, 10),
```

Comment:
This is just a small naming detail but maybe simply `formId` would be nicer considering the rendered code (`<!-- wp:mailpoet/form-block {"selectedForm":1} /-->`).